pkg/controller/master/nodedrain/nodedrain_controller.go

```
@@ -77,121 +77,120 @@ func (ndc *ControllerHandler) OnNodeChange(_ string, node *corev1.Node) (*corev1
 	if node == nil || node.DeletionTimestamp != nil {
 		return node, nil
 	}
-	_, ok := node.Annotations[drainhelper.DrainAnnotation]
 
+	_, ok := node.Annotations[drainhelper.DrainAnnotation]
 	if !ok {
 		return node, nil
 	}
 
 	_, forced := node.Annotations[drainhelper.ForcedDrain]
 
-	if ok {
-		// still running a check in the background to avoid maintenance issues when using object annotations
-		// directly
-		err := drainhelper.DrainPossible(ndc.nodeCache, node)
-		if err != nil {
-			return node, err
-		}
+	// still running a check in the background to avoid maintenance issues when using object annotations
+	// directly
+	err := drainhelper.DrainPossible(ndc.nodeCache, node)
```

It is not directly related to this PR, but `DrainPossible` seems to have issues:
(1) If it failed to list things, then return error is okay for retry
(2) if it failed due to error like `errSingleControlPlaneNode` and `errHAControlPlaneNode`, the whole drain (or the maintenance mode) should be put to failure instead of return error to reconciller
 
 `DrainPossible` is more like `IsDrainPossible`,  return `bool, err` will not lead such

---

Done